streamer/src/nonblocking/recvmmsg.rs

```diff
@@ -36,6 +38,21 @@ pub async fn recv_mmsg(
     Ok(i)
 }
 
+/// Reads the exact number of packets required to fill `packets`
+pub async fn recv_mmsg_exact(
+    socket: &UdpSocket,
+    packets: &mut [Packet],
+) -> io::Result</*num packets:*/ usize> {
+    let total = packets.len();
+    let mut remaining = total;
+    while remaining != 0 {
+        let first = total - remaining;
+        let res = recv_mmsg(socket, &mut packets[first..]).await?;
+        remaining -= res;
+    }
+    Ok(total)
```

Comment:
Nit: this looks like it's always going to be Ok(packets.len()) if this function succeeds... Do we want this or should we either get rid of the usize, or return the number of packets read on error as well?

Reply:
I was thinking about mimicking behavior of https://docs.rs/tokio/latest/tokio/io/trait.AsyncReadExt.html#method.read_exact
I briefly looked into their implementation and it looks like they always return all or nothing: https://docs.rs/tokio/latest/src/tokio/io/util/read_exact.rs.html#65
I think `total => packets.len()` might be fine